tests/bson/bson-utcdatetimeinterface-002.phpt

```diff
@@ -0,0 +1,105 @@
+--TEST--
+MongoDB\BSON\UTCDateTimeInterface example with MongoDB\BSON\TypeWrapper
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+class MyUTCDateTime implements MongoDB\BSON\TypeWrapper, MongoDB\BSON\UTCDateTimeInterface
+{
+    private $date;
+
+    public function __construct($milliseconds)
+    {
+        $this->date = new MongoDB\BSON\UTCDateTime($milliseconds);
+    }
+
+    public static function createFromBSONType(MongoDB\BSON\Type $type)
+    {
+        if ( ! $type instanceof MongoDB\BSON\UTCDateTime) {
+            throw new InvalidArgumentException('Cannot create MyUTCDateTime from ' . get_class($type));
+        }
+
+        return new self((string) $type);
```

Comment:
We discussed this while doing the HHVM versions, and this is what the conclusion was I believe: https://github.com/mongodb/mongo-hhvm-driver/blob/master/ext_mongodb.php#L1182